client/solve.go

```diff
@@ -505,3 +531,38 @@ func parseCacheOptions(ctx context.Context, isGateway bool, opt SolveOpt) (*cach
 	}
 	return &res, nil
 }
+
+// getOCILayoutStores look for --opt content:<image>=oci-layout://...@sha256:... options.
+// This func's job is to find the content.Store path, open it, and pass a reference to it,
+// identifying the content.Store via a unique identifier. We don't actually care about the
+// image or the hash here, since getOCILayoutStores's job is only to get unique idnetifiers
+// for each store pointing to a content.Store.
+func getOCILayoutStores(ctx context.Context, opt SolveOpt) (map[string]content.Store, error) {
+	const (
+		contextPrefix   = "context:"
+		ociLayoutPrefix = "oci-layout://"
```

Comment:
`client` package should not parse `FrontendAttrs` map. These are the string config passed to frontend. Just add `OCIStores map[string]content.Store` to the `SolveOpt` type.

Reply:
Done. Basically moved that parsing logic to `cmd/buildctl/build.go`, where it parses the CLI flags anyways, leaving the `client` package to just have the map to work with.

I am not sure that it makes proper sense now, though.

What should the key of the `map[string]content.Store` be? When it is mapping, e.g. caches for local in `client`, it is `"local:"+dir`. I am not sure that translates correctly to `cmd/buildctl`, and it splits that convention between them.

More importantly, the opt actually should be mapping not just the dir, but the image. So if I have:

```
--opt context:foo=some/dir@sha256:abcd --opt context:bar=some/other@sha256ffcc
```

Then it isn't just about having the store ID point to the correct store (e,g. `oci-layout:some/dir`), but knowing that `foo` points there to make it useful.

Or did we already cover that elsewhere?

Reply:
For `buildctl` it is cleaner to make this same as `--local` where to use named context you would atm do `buildctl build --local foo=./path/to/dir --opt context:foo=local://foo` . The key is whatever is the key in the new flag. So I guess `--oci-layout foo=./path@sha256:abc`

Reply:
Update to previous comment, I think you prefer digest to be part of opt. `buildctl build --oci-layout foo=./path/to/dir --opt context:alpine=oci-layout://foo@sha256:abc`

Reply:
Shouldn't he two be the same? Aren't these conceptually the same thing, just one is a convenience?

* `--local foo=/path/to/dir`
* `--opt context:foo=local:///path/to/dir`

Similarly, wouldn't these be identical?

* `--oci-layout bar=/path/to/dir@sha256:abcdef`
* `--opt context:bar=oci-layout:///path/to/dir@sha256:abcdef`

I didn't quite do the `--oci-layout` convenience yet. If this is correct, I am fine adding it. Let me know.